bftengine/src/preprocessor/messages/ClientPreProcessRequestMsg.cpp

```diff
@@ -10,18 +10,27 @@
 // file.
 
 #include "ClientPreProcessRequestMsg.hpp"
+#include "SimpleClient.hpp"
 #include "assertUtils.hpp"
 
 namespace preprocessor {
 
 using namespace std;
 using namespace bftEngine;
 
+ClientPreProcessRequestMsg::ClientPreProcessRequestMsg(NodeIdType sender,
+                                                       uint64_t reqSeqNum,
+                                                       uint32_t requestLength,
+                                                       const char* request)
+    : ClientRequestMsg(sender, PRE_PROCESS_REQ, reqSeqNum, requestLength, request) {
+  msgBody_->msgType = MsgCode::ClientPreProcessRequest;
+}
+
 void ClientPreProcessRequestMsg::validate(const ReplicasInfo& repInfo) const {
   Assert(type() == MsgCode::ClientPreProcessRequest);
   Assert(senderId() != repInfo.myId());
 
-  if (size() < (sizeof(ClientRequestMsgHeader) + msgBody()->requestLength))
+  if (size() < sizeof(ClientRequestMsgHeader) || size() < (sizeof(ClientRequestMsgHeader) + msgBody()->requestLength))
```

Comment:
Why do we need this additional check?
Actually `size() < sizeof(ClientRequestMsgHeader)` implies `size() < (sizeof(ClientRequestMsgHeader) + msgBody()->requestLength)`

Reply:
Ha, I'll forward related Guy's mail to you.

Reply:
Maybe add a comment indicating why, or wrap these checks in a template function with a header comment, since they are duplicated.